pandas/core/dtypes/inference.py

```diff
@@ -282,12 +285,20 @@ def is_list_like(obj):
     >>> is_list_like(np.array(2)))
     False
     """
-
-    return (isinstance(obj, compat.Iterable) and
-            # we do not count strings/unicode/bytes as list-like
-            not isinstance(obj, string_and_binary_types) and
-            # exclude zero-dimensional numpy arrays, effectively scalars
-            not (isinstance(obj, np.ndarray) and obj.ndim == 0))
```

Comment:
Aside from adding the kwarg everywhere, this is the only substantial change of this PR.